repos/system_upgrade/el7toel8/files/rhel_upgrade.py

```diff
@@ -48,13 +48,42 @@ def _process_packages(self, pkg_set, op):
                        'in repositories metadata: '.format(op.__name__) + ' '.join(pkgs_notfound))
             print('Warning: ' + err_str, file=sys.stderr)
 
+    def _save_aws_region(self, region):
+        self.plugin_data['rhui']['aws']['region'] = region
+        with open(self.opts.filename, 'w+') as fo:
+            json.dump(self.plugin_data, fo, sort_keys=True, indent=2)
+
+    def _read_aws_region(self, repo):
+        region = None
+        if repo.baseurl:
+            region = repo.baseurl.split('.', 2)[1]
+        elif repo.mirrorlist:
+            region = repo.mirrorlist.split('.', 2)[1]
+        if not region:
+            print('Could not read AWS REGION from either baseurl or mirrorlist', file=sys.stderr)
+            sys.exit(1)
+        return region
+
+    def _fix_rhui_url(self, repo, region):
+        if repo.baseurl:
+            repo.baseurl = tuple(
+                url.replace('REGION', region, 1) for url in repo.baseurl
+            )
+        elif repo.mirrorlist:
+            repo.mirrorlist = repo.mirrorlist.replace('REGION', region, 1)
+        else:
+            raise dnf.exceptions.RepoError("RHUI repository %s does not have an url" % repo.name)
+        return repo
+
     def pre_configure(self):
         with open(self.opts.filename) as fo:
             self.plugin_data = json.load(fo)
         # There is an issue that ignores releasever value if it is set at configure
         self.base.conf.releasever = self.plugin_data['dnf_conf']['releasever']
 
     def configure(self):
+
+        on_aws = bool(self.plugin_data['rhui']['aws']['on_aws'])
```

Comment:
hmm... why the `bool()` here? I mean, what values are possible inside `self.plugin_data['rhui']['aws']['on_aws']`? It should be already boolean `True`/`False` right?